Example/DKHelperTests/NSDateTests.swift

```diff
@@ -0,0 +1,292 @@
+		XCTAssertNotNil(date)
+		XCTAssertEqual(date?.year(), 1992)
+		XCTAssertEqual(date?.day(), 24)
+		XCTAssertEqual(date?.month(), 12)
+	}
+
+	func testShouldReturnNilDate() {
+		let date = NSDate(fromString: "Dec 24, 1992", style: .FullStyle)
+		XCTAssertNil(date)
+	}
+
+	func testShouldReturnValidAndCorrectDateWithStringFormat() {
+		let date = NSDate(fromString: "24.12.1992", format: "dd.MM.yyyy")
+		XCTAssertNotNil(date)
+		XCTAssertEqual(date?.year(), 1992)
+		XCTAssertEqual(date?.day(), 24)
+		XCTAssertEqual(date?.month(), 12)
+	}
+
+	func testShouldReturnValidTodayDate() {
+		if let today = self.today {
+			let date = NSDate(fromISOString: today.ISO8601StringValue())
+			XCTAssertNotNil(date)
+			XCTAssertEqual(date?.year(), today.year())
+			XCTAssertEqual(date?.day(), today.day())
+			XCTAssertEqual(date?.month(), today.month())
+			XCTAssertEqual(date?.hour(), today.hour())
+			XCTAssertEqual(date?.minute(), today.minute())
+			XCTAssertEqual(date?.second(), today.second())
+		} else {
+			XCTFail()
+		}
+	}
+
+	func testReturnValidStringValueOfTodayDate() {
+		if let today = self.today {
+			let value = today.stringValue()
+			XCTAssertNotNil(value)
+			XCTAssertTrue(value.characters.count > 0)
+		} else {
+			XCTFail()
+		}
+	}
+}
+
+// MARK: - Getting Date Components
+
+extension NSDateTests {
+
+	func testShouldReturnDatesComponents() {
+		let date = NSDate(timeIntervalSince1970: 0)
+		XCTAssertEqual(date.year(), 1970)
+		XCTAssertEqual(date.day(), 1)
+		XCTAssertEqual(date.month(), 1)
+		XCTAssertEqual(date.hour(), 1, "expect Time Zone GMT+1")
+		XCTAssertEqual(date.minute(), 0)
+		XCTAssertEqual(date.second(), 0)
+	}
+
+	func testShouldReturnTodaysMidnightDate() {
+		let date = today?.midnightDate()
+		XCTAssertNotNil(date)
+		XCTAssertEqual(date?.year(), today?.year())
+		XCTAssertEqual(date?.day(), today?.day())
+		XCTAssertEqual(date?.month(), today?.month())
+		XCTAssertEqual(date?.hour(), 1, "expect Time Zone GMT+1")
+		XCTAssertEqual(date?.minute(), 0)
+		XCTAssertEqual(date?.second(), 0)
+	}
+
+	func testShouldReturnTodaysDayDate() {
+		let date = NSDate.currentDayDate()
+		XCTAssertNotNil(date)
+		XCTAssertEqual(date?.year(), today?.year())
+		XCTAssertEqual(date?.day(), today?.day())
+		XCTAssertEqual(date?.month(), today?.month())
+		XCTAssertEqual(date?.hour(), 1, "expect Time Zone GMT+1")
+		XCTAssertEqual(date?.minute(), 0)
+		XCTAssertEqual(date?.second(), 0)
+	}
+}
+
+// MARK: - Date dispaly Strings
+
+extension NSDateTests {
+
+	func testShouldReturnValidAndCorrectDateComponentNaming() {
+		let date = NSDate(fromString: "04.12.2015", format: "dd.MM.yyyy")
+		let dayName = date?.dayName()
+		let monthName = date?.monthName()
+		XCTAssertNotNil(date)
+		XCTAssertNotNil(dayName)
+		XCTAssertNotNil(monthName)
+		XCTAssertEqual(dayName, "Freitag", "expect german localization")
+		XCTAssertEqual(monthName, "Dezember", "expect german localization")
+	}
+
+	func testShouldReturnValidAndCorrectDateDisplayStrings() {
```

Comment:
please rename the function `test_Should...` to be consistent with the other files.